### DISCUSSION TITLE
[@types/markdown-it] Core rule function should return `void`

### DISCUSSION THREAD

**Discussion by Lemmingh** [DefinitelyTyped/DefinitelyTyped#56245] [Issues with a @types package] [answered]
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f346df47bf74d9c5572343c97d58c4b2a632323b/types/markdown-it/lib/parser_core.d.ts#L7>

However, a `core` rule function is expected to return `void`, that is to say,

```typescript
type RuleCore = (state: StateCore) => void;
```

according to

<https://github.com/markdown-it/markdown-it/blob/6e2de08a0b03d3d0dcc524b89710ce05f83a0283/lib/parser_core.js#L40-L53>

Others are correct.

[a comment by an automated account is not shown]

**Reply by peterblazejewicz** [reply]
fixing #56249
